services/bettersocial/post/createPostV3.js

```diff
@@ -284,6 +331,14 @@ const BetterSocialCreatePostV3 = async (req, isAnonimous = true) => {
       });
     }
 
+    await processSendSystemMessage(
+      filteredTopics,
+      req?.userId,
+      user?.username,
+      body?.with_system_message,
+      isAnonimous
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `processSendSystemMessage` is called with `req?.userId` and `user?.username`. It's important to ensure that these values are not null or undefined before calling the function, as they are required parameters. If they are missing, an error will be thrown inside `processSendSystemMessage`, which could cause unexpected behavior.

```diff
-    await processSendSystemMessage(
-      filteredTopics,
-      req?.userId,
-      user?.username,
-      body?.with_system_message,
-      isAnonimous
-    );
+    if (req?.userId && user?.username) {
+      await processSendSystemMessage(
+        filteredTopics,
+        req.userId,
+        user.username,
+        body?.with_system_message,
+        isAnonimous
+      );
+    } else {
+      logger.warn('Missing userId or username, skipping send system message');
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->